pkg/manifests/config.go

```diff
@@ -291,9 +291,8 @@ func (c *Config) applyDefaults() {
 	if c.ClusterMonitoringConfiguration.K8sPrometheusAdapter == nil {
 		c.ClusterMonitoringConfiguration.K8sPrometheusAdapter = &K8sPrometheusAdapter{}
 	}
-	if c.ClusterMonitoringConfiguration.K8sPrometheusAdapter.DedicatedServiceMonitors == nil {
-		c.ClusterMonitoringConfiguration.K8sPrometheusAdapter.DedicatedServiceMonitors = &DedicatedServiceMonitors{}
-	}
+	// TODO: Remove this in 4.XX?
```

Comment:
waiting for decision https://github.com/openshift/cluster-monitoring-operator/pull/2210#issuecomment-1863835566

(specify a version or get rid of this + its doc)

Reply:
My recommendation would be:
* remove the field from the documentation
* keep the field in the config struct "forever"
* have a metric + alert (info severity) in CMO reporting that the field is in use but has no effect.

Reply:
Works for me, happy to defers to Simons experience here :)
I suppose we don't have a metric like that yet? Maybe something like `cluster_monitoring_operator_deprecated_config_use{deprecated_field="$FIELD_NAME"}`? Is it worth including the version this was deprecated?

Reply:
We'll be able to get rid of it during the migration into the CRD :)

I'd say that a log (to ease any potential debugging related to the config) + the deprecation notice  in the release notes are sufficient especially that the change should be transparent to users. If users have nothing to worry about (config kept forever), why warn them?

We could give the metric a try, it could be helpful during the migration to CRD, but once the CRD are in place, it'd be easier for CMO to prune those deprecated fields.

Reply:
Yes, we could add the deprecation version as well, I'll give it a try.